Automattic/wp-calypso
# Editor Checkout: E2E tests for complete flow of upgrading from free to premium plan via editor

#### Changes proposed in this Pull Request

Additional E2E tests for the in-editor checkout covers the following scenarios:

* User has a free plan and can view the checkout modal from clicking the Jetpack upgrade button on a premium block
* User has the ability to edit the orders plan length
* User has the ability to successfully apply a coupon code
* User can fill out card details, unless they have saved card details
* User successfully has payment processed and declines up-sell offer
* User lands back into the same post they were editing
* User can successfully delete the plan on their account

**Other changes:**
* I have also added the ability to set the cookie to use the payments sandbox when using the login flow.

#### Testing instructions (Video below)

* Add, and then remove the "[Status] Needs e2e Testing" label. This will trigger CircleCI to run the test suites again.
* After the tests have run, click "Details" on `ci/wp-e2e-tests-full-desktop` below.
* Click "Steps" tab and then locate the `wp-calypso-gutenberg-checkout-spec.js` test file in the tests and check they've passed.
* All tests in this spec should pass, and there should be no Editor Checkout related test failures in the "Tests" tab.

[Video](https://www.loom.com/share/87300b6bffd74570a9426d81c8e46b2f)

Fixes https://github.com/Automattic/wp-calypso/issues/46365
